src/coreclr/tools/aot/ILCompiler.LLVM/CodeGen/LLVMObjectWriter.cs

```diff
@@ -182,184 +155,143 @@ public static void EmitObject(string objectFilePath, IEnumerable<DependencyNode>
-                LLVMValueRef valRef = module.GetNamedAlias(SymbolName);
-                if (valRef.Handle == IntPtr.Zero)
-                {
-                    valRef = module.GetNamedFunction(SymbolName);
-                }
-                Debug.Assert(valRef.Handle != IntPtr.Zero, $"Undefined symbol: {SymbolName}");
-
-                if (Offset != 0)
-                {
-                    LLVMValueRef[] index = new[] { LLVMValueRef.CreateConstInt(LLVMTypeRef.Int32, Offset) };
-                    valRef = LLVMValueRef.CreateConstGEP2(LLVMTypeRef.Int8, valRef, index);
-                }
-
-                return valRef;
-            }
-        }
-
-        private struct ObjectNodeDataEmission
-        {
-            public ObjectNodeDataEmission(LLVMValueRef node, byte[] data, Dictionary<int, SymbolRefData> objectSymbolRefs)
-            {
-                _node = node;
-                _data = data;
-                _objectSymbolRefs = objectSymbolRefs;
-            }
-            private LLVMValueRef _node;
-            private readonly byte[] _data;
-            private readonly Dictionary<int, SymbolRefData> _objectSymbolRefs;
-
-            public void Fill(LLVMObjectWriter writer)
-            {
-                int pointerSize = writer._nodeFactory.Target.PointerSize;
-                ArrayBuilder<LLVMValueRef> entries = default;
-                int countOfPointerSizedElements = _data.Length / pointerSize;
-
-                for (int i = 0; i < countOfPointerSizedElements; i++)
-                {
-                    int curOffset = (i * pointerSize);
-                    SymbolRefData symbolRef;
-                    if (_objectSymbolRefs.TryGetValue(curOffset, out symbolRef))
-                    {
-                        entries.Add(symbolRef.ToLLVMValueRef(writer._module));
-                    }
-                    else
-                    {
-                        ulong value = (pointerSize == 4) ? BitConverter.ToUInt32(_data, curOffset)
-                                                         : BitConverter.ToUInt64(_data, curOffset);
-                        entries.Add(LLVMValueRef.CreateConstIntToPtr(LLVMValueRef.CreateConstInt(writer._intPtrType, value), writer._ptrType));
-                    }
-                }
-
-                _node.Initializer = LLVMValueRef.CreateConstArray(writer._ptrType, entries.ToArray());
-            }
-        }
-
         private void FinishObjWriter()
         {
-            // Since emission to llvm is delayed until after all nodes are emitted... emit now.
-            foreach (var nodeData in _dataToFill)
-            {
-                nodeData.Fill(this);
-            }
-
             EmitKeepAliveList();
 
 #if DEBUG
             _module.PrintToFile(Path.ChangeExtension(_objectFilePath, ".txt"));
-#endif
             _module.Verify(LLVMVerifierFailureAction.LLVMAbortProcessAction);
+
+            _module.PrintToFile(Path.ChangeExtension(_objectFilePath, "external.txt"));
```

Comment:
Is it intentional to change extension to `external.txt` here? `external.txt` does not look like an extension.

Reply:
The intent is to transform `<path>/Name.bc` (`_objectFilePath`) into `<path>/Name.external.txt` (this pattern is used in a couple other places). Using `Path.ChangeExtension` a little bit of an abuse, but the alternative is open-coding it.